pkg/endpointmanager/manager.go

```diff
@@ -74,24 +75,51 @@ func NewEndpointManager(epSynchronizer EndpointResourceSynchronizer) *EndpointMa
 	return &mgr
 }
 
+// waitForProxyCompletions blocks until all proxy changes have been completed.
+func (mgr *EndpointManager) waitForProxyCompletions(proxyWaitGroup *completion.WaitGroup) error {
+	err := proxyWaitGroup.Context().Err()
+	if err != nil {
+		return fmt.Errorf("context cancelled before waiting for proxy updates: %s", err)
+	}
+
+	start := time.Now()
+	log.Debug("Waiting for proxy updates to complete...")
+	err = proxyWaitGroup.Wait()
+	if err != nil {
+		return fmt.Errorf("proxy updates failed: %s", err)
+	}
+	log.Debug("Wait time for proxy updates: ", time.Since(start))
+
+	return nil
+}
+
 // UpdatePolicyMaps returns a WaitGroup which is signaled upon once all endpoints
 // have had their PolicyMaps updated against the Endpoint's desired policy state.
-func (mgr *EndpointManager) UpdatePolicyMaps() *sync.WaitGroup {
+func (mgr *EndpointManager) UpdatePolicyMaps(ctx context.Context) *sync.WaitGroup {
 	var wg sync.WaitGroup
 
+	proxyWaitGroup := completion.NewWaitGroup(ctx)
+
 	eps := mgr.GetEndpoints()
-	wg.Add(len(eps))
+	wg.Add(len(eps) + 1)
```

Comment:
It might be worth explaining that the +1 is for the proxy further down. Not a big deal, though

Reply:
Done.